Reorganize Tania monorepo

bepitulaz opened the pull request:
I want to rearrange the project structure for better separation of concern before doing anymore development based on the roadmap. The structure will be:

```
|- build.sh
|- readme.md
|- backend
|---- main.go
|---- go.mod
|---- conf.json
|---- other_backend_src
|- dist (build directory)
|- frontend
|---- languages
|-------- locale
|---- storybook
|---- package.json
|---- other_frontend_src
|- documentation
|---- postman
```

This PR will also change the build script because Tania won't be maintained by Tanibox. Thus, all build infrastructure to automatically generate cross platform binary is turned off.

### Notes

The old frontend inside `frontend-vue` is not usable, because many obsolete packages. It even can't be built by using `yarn dev`. I won't bother with it anymore, and continue to integrate this new [Next.js (React)](https://github.com/usetania/tania-core/tree/2.0-dev/frontend) frontend into the Go backend.

bepitulaz committed:
Moving all Go codes to backend directory.

bepitulaz committed:
Creating a script to build the Golang code.

bepitulaz committed:
Fixing Github action for golangci-lint.

bepitulaz committed:
Remove only-new-issues in golangci-lint because it's not compatible with working-directory.

pablojimpas commented:
This was very much needed! I'll review it more in depth once it's ready but so far this proposed directory layout looks good to me.

bepitulaz committed:
Moving the old front-end to its own folder.

bepitulaz commented:
@pablojimpas Please, when you are free, you can start review this PR 😊

pablojimpas commented:
> @pablojimpas Please, when you are free, you can start review this PR blush

Okey, I'll left more specific comments little by little in the coming days.

For now I think we might encounter some issues with the build scripts as well as config and database files. We have to be really careful with relative paths and assumptions that were made before.

I'll like to further streamline the first hierarchy level, the `languages` folder only seems to be relevant to the frontend and the same goes for `postman`, `configuration` and `database` folders for the backend. I don't like the `jsonschema` folder in the first level either.

Maybe it's best trying to get rid of the scripts if possible and just have a top level `Makefile`. Furthermore, I think we can nuke the old Vue frontend in this branch since it will no longer be developed, we can always take it as a reference in the old branch.

pablojimpas commented on scripts/build.sh:
```diff
@@ -0,0 +1,21 @@
+#/bin/bash
```
Use #!, not just #, for the shebang. Also, consider using `#!/bin/sh` instead of `bash` to eliminate the dependency on it and make it more portable across POSIX-like operating systems.

pablojimpas commented on scripts/build.sh:
```diff
@@ -0,0 +1,21 @@
+
+# Build the binary
+echo "Building golang binaries..."
+go build -o ../dist/tania main.go
```
The last `main.go` could be removed to make it less explicit and avoid future issues if something changes.

pablojimpas commented on backend/main.go:
```diff
@@ -252,7 +252,7 @@ func initMysql() *sql.DB {
 	log.Print("Using MySQL at ", host, ":", port, "/", dbname)
 
-	ddl, err := ioutil.ReadFile("db/mysql/ddl.sql")
+	ddl, err := ioutil.ReadFile("database/mysql/ddl.sql")
```
This will fail when running the backend in development since the database directory is one level up.

pablojimpas commented on backend/main.go:
```diff
@@ -309,7 +309,7 @@ func initSqlite() *sql.DB {
 	log.Print("Executing DDL file for ", *config.Config.SqlitePath)
 
-	ddl, err := ioutil.ReadFile("db/sqlite/ddl.sql")
+	ddl, err := ioutil.ReadFile("database/sqlite/ddl.sql")
```
Same

bepitulaz committed:
Delete the old frontend

bepitulaz commented:
Now, I change the structure again:
- Deleted the old Vue frontend
- Added the `documentation` folder for `postman` and `jsonschema`, because these are not part of the backend code.
- Moved `languages` to the frontend
- Moved `database` and `conf.json` inside the `backend` folder, so it won't fail in the development mode
- Use single build script at the root of the project.

bepitulaz committed:
Remove logging the user input that hasn't sanitized.

bepitulaz committed:
Updated Readme file.

bepitulaz commented:
Alright, I'm done with this branch @pablojimpas. I'll keep this PR open until next week, in case any feedback related to this branch.

pablojimpas commented:
> Alright, I'm done with this branch @pablojimpas. I'll keep this PR open until next week, in case any feedback related to this branch.

Perfect. I'll review it in my spare time but so far the revised changes look good!

wotnak commented:
Some issues in the README that may be worth fixing:
- image file for the logo at the top of the README is missing,
- the ToC section has 'Run Tania Core On Portainer' element that links to not existing section, you may also consider removing the whole ToC section as currently github [provides its functionality](https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/) built in into the ui,
- 'chat on Telegram' link doesn't seem to be working, trying to use it results in an error message 'User does not exist',
- 'Prerequisites' section doesn't seem to be in sync with what versions are currently required by frontend and backend,
- there are two 'Roadmap' sections linking to different places, one of them should be probably removed.

bepitulaz commented:
Thanks for pointing it @wotnak. I didn't realize that GitHub has the built-in ToC functionality.

bepitulaz committed:
Updating the Readme file.

bepitulaz committed:
Remove Telegram link from the Readme.

bepitulaz committed:
Add new logo of Tania.
